packages/x-components/src/x-modules/tagging/wiring.ts

```diff
@@ -81,7 +81,16 @@ export const setSessionDuration = wireCommit('setSessionDuration');
 export const trackQueryWire = moduleDebounce(
   wireDispatch('track'),
   ({ state }) => state.config.queryTaggingDebounceMs,
-  { cancelOn: 'UserClearedQuery' }
+  {
+    cancelOn: 'UserClearedQuery',
+    forceOn: [
+      'UserClickedAResult',
+      'UserClickedAPromoted',
+      'UserClickedABanner',
+      'UserClickedARedirection',
+      'UserReachedResultsListEnd'
+    ]
+  }
```

Comment:
This is the actual change, the rest are test related changes. I think I've added any event that might override the query tagging while still consider the search response as accepted.